aws/resource_aws_opsworks_chef.go

```diff
@@ -0,0 +1,335 @@
+				Default:  false,
+			},
+
+			// TODO:
+			// I kinda want to make this required because how else are we going to get at that info?
+			// Also, since these are write-only and not returned by the API, we'll probably need a custom diff function that ignores them?
+			"chef_pivotal_key": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Default:   nil,
+				Sensitive: true,
+			},
+
+			"chef_delivery_admin_password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Default:   nil,
+				Sensitive: true,
+			},
+
+			// TODO:
+			// default and only valid. I'll leave it configurable for future proofing but
+			// validation failure, maybe?
+			"engine_model": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  EngineModel,
+			},
+
+			// TODO:
+			// default and only valid. I'll leave it configurable for future proofing but
+			// validation failure, maybe?
+			"engine_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  EngineVersion,
+			},
+
+			// TODO: document the instance role required
+			// https://s3.amazonaws.com/opsworks-cm-us-east-1-prod-default-assets/misc/opsworks-cm-roles.yaml
+			"instance_profile_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"ssh_key_pair": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"preferred_backup_window": {
+				Type:     schema.TypeString,
+				Required: true, // TODO: ?
+			},
+
+			"preferred_maintenance_window": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// TODO:
+			// if you don't specify this, it'll create one. I'm of the opinion that users should
+			// create it themselves. Or maybe some sort of wrapper module? I dunno.
+			// if it's optional, will they be computed? if so, is that something we can express here?
+			"security_group_ids": {
+				Type:     schema.TypeList, // TODO: should this be a schema.TypeSet instead so order doesn't matter? check ALB and aws_instance resources?
+				Required: true,
+			},
+
+			// TODO: document this role creation
+			// https://s3.amazonaws.com/opsworks-cm-us-east-1-prod-default-assets/misc/opsworks-cm-roles.yaml
+			"service_role_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// TODO: huh?
+			// The IDs of subnets in which to launch the server EC2 instance.
+			//
+			// Amazon EC2-Classic customers: This field is required. All servers must run
+			// within a VPC. The VPC must have "Auto Assign Public IP" enabled.
+			//
+			// EC2-VPC customers: This field is optional. If you do not specify subnet IDs,
+			// your EC2 instances are created in a default subnet that is selected by Amazon
+			// EC2. If you specify subnet IDs, the VPC must have "Auto Assign Public IP"
+			// enabled.
+			// if it's optional, will they be computed? if so, is that something we can express here?
```

Comment:
All sorts of :wat: here.

I'm thinking we should force people to specify subnet IDs. But that's just me.